internal/provider/models/sources/test/open_telemetry_metrics_test.go

```diff
@@ -0,0 +1,195 @@
+package sources
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	. "github.com/mezmo/terraform-provider-mezmo/internal/provider/providertest"
+)
+
+type resourceOpenTelemetryMetricsSourceConfig struct {
+	PipelineName    string
+	PipelineTitle   string
+	SourceName      string
+	SourceTitle     string
+	SourceDesc      string
+	CaptureMetadata bool
+
+	// Will include a shared source which means setting the gateway_route_id on
+	// source A to the ID of source B.
+	GatewayResourceID string
+}
+
+var resourceOpenTelemetryMetricsSourceConfigTpl = `
+resource "mezmo_pipeline" "{{.PipelineName}}" {
+	title = "{{.PipelineTitle}}"
+}
+
+resource "mezmo_open_telemetry_metrics_source" "{{.SourceName}}" {
+	pipeline_id = mezmo_pipeline.{{.PipelineName}}.id
+	title       = "{{.SourceTitle}}"
+	description = "{{.SourceDesc}}"
+
+	{{if .CaptureMetadata}}
+	capture_metadata = "{{.CaptureMetadata}}"
+	{{end}}
+}
+
+{{if .GatewayResourceID}}
+resource "mezmo_open_telemetry_metrics_source" "{{.SourceName}}-gateway" {
+	pipeline_id      = mezmo_pipeline.{{.PipelineName}}.id
+	gateway_route_id = {{.GatewayResourceID}}
+}
+{{end}}
+`
+
+func TestAccOpenTelemetryMetricsSource(t *testing.T) {
+	pipelineName := "test_parent"
+	pipelineTitle := "parent pipeline"
+	sourceName := "my_source"
+	sourceTitle := "my open telemetry metrics title"
+	sourceDesc := "my open telemetry metrics description"
+
+	resourceName := fmt.Sprintf("mezmo_open_telemetry_metrics_source.%s", sourceName)
+	resourceNamePipeline := fmt.Sprintf("mezmo_pipeline.%s", pipelineName)
+	resourceNameGateway := fmt.Sprintf("mezmo_open_telemetry_metrics_source.%s-gateway", sourceName)
```

Comment:
Hmm, I'd argue that this isn't easier or clearer.  It provides a layer of indirection that I find harder to follow than inlined definitions like we do in other tests.  Here, you have to defined the vars, call the template parser, match the struct field names up to your variables, and catch errors.

May I ask what you found cumbersome about the inline definitions?  Usually, the use of `SetCacheConfig` and `GetCachedConfig` is enough to not have to repeat the basic elements, leaving the test to only inline the thing you're testing.  In this case, the `OpenTelemetryMetricsSourceModel` is pretty basic.

Reply:
I found it hard to reason about what config each test case was passed, but maybe that's just me. The idea was to combine all the Otel sources into one acceptance test later which I think is easier to do with the template since the same string can be resued across all 3 sources.

Sorry for cluttering it a bit but I just added the logs source and a common test case for all 3 Otel sources, I hope that makes the intent behind the change clearer.

Let me know what you think about it now.